vehicle/remote_control_process.py

```diff
@@ -246,1086 +306,939 @@ def close_motor_socket(self):
+                                 f"List length {len(self.nav_path_list)}, "
+                                 f"Closest path index {closest_row_index}, "
+                                 f"min_distance {min_distance}")
+                self.nav_path_index = closest_row_index
+
+            self.load_path(self.nav_path_list[self.nav_path_index],
+                           simulation_teleport=True, generate_spline=True)
+            self.reloaded_path = True
 
-                time6 = time.time() - debug_time
+        else:
+            self.load_path(self.robot_object.loaded_path,
+                           simulation_teleport=True, generate_spline=True)
+            self.reloaded_path = True
+        self.load_path_time = time.time()
 
-                # Get joystick value
-                if self.simulated_hardware and False:
-                    joy_steer, joy_throttle, joy_strafe = 0.0, 0.0, 0.0
+    def steering_calc(self):
+        """
+        Begin steering calculation.
+        TODO: Break this out to a separate module.
+        """
+
+        if not self.nav_path.points:
+            return (gps_tools.GpsPoint(0, 0),  # projected_path_tangent_point,
+                    None,  # closest_path_point,
+                    math.inf,  # absolute_path_distance,
+                    None,  # drive_reverse,
+                    None,  # calculated_rotation,
+                    None)  # calculated_strafe)
+
+        closest_path_point, path_point_heading = self.calc_closest_path_point_and_path_point_heading()
+        absolute_path_distance = gps_tools.get_distance(self.gps.last_sample(), closest_path_point)
+        calculated_rotation = path_point_heading - self.gps.last_sample().azimuth_degrees
+        # Truncate values to between 0 and 360
+        calculated_rotation %= 360
+        # Set value to +/- 180
+        if calculated_rotation > 180:
+            calculated_rotation -= 360
+        self.logger.debug("calculated_rotation: {}, DISTANCE: {}".format(
+            calculated_rotation, abs(absolute_path_distance)))
+        self.logger.debug("robot heading {}, path heading {}".format(
+            self.gps.last_sample().azimuth_degrees, path_point_heading))
+
+        projected_path_tangent_point = gps_tools.project_point(closest_path_point, path_point_heading, 3.0)
+        gps_lateral_distance_error = gps_tools.get_approx_distance_point_from_line(
+            self.gps.last_sample(), closest_path_point, projected_path_tangent_point)
+        calculated_strafe = gps_lateral_distance_error
+
+        _MAXIMUM_ROTATION_ERROR_DEGREES = 140
+
+        vehicle_dir = self.nav_path.navigation_parameters.vehicle_travel_direction
+        path_dir = self.nav_path.navigation_parameters.path_following_direction
+        drive_solution_okay = (vehicle_dir == Direction.EITHER or
+                               vehicle_dir == path_dir and abs(calculated_rotation) <= _MAXIMUM_ROTATION_ERROR_DEGREES)
```

Comment:
Note the style change here. Instead of setting variables in nested `if..elif..`s, each variable is set independently. Reasoning about the logic is a little easier in this way.